stages/org.osbuild.xz

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python3
+"""
+Compress a file
+
+Buildhost commands used: `xz`.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+
+
+SCHEMA_2 = r"""
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["file"],
+  "properties": {
+    "file": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+},
+"options": {
+  "additionalProperties": false,
+  "required": ["filename"],
+  "properties": {
+    "filename": {
+      "description": "Filename to use for the compressed file",
+      "type": "string"
+    }
+  }
+}
+"""
+
+
+def parse_input(inputs):
+    image = inputs["file"]
+    files = image["data"]["files"]
```

Comment:
Schema doesn't seem to enforce these keys.  Should they be checked instead of reading directly?

Actually, looking at the stage test for `xz`, I'm not entirely sure how this works.  What am I missing?

Reply:
I think I need to somehow document this since it seems to be a constant source of confusion. Only the `options` in the arguments dict is coming from the manifest, everything else is assembled by `osbuild` itself: https://github.com/osbuild/osbuild/blob/88da3beabc75ccff7733abe41d8e08f418d5be00/osbuild/pipeline.py#L102-L116

The actual contents of `inputs["file"]`, is, in this case, coming from the `org.osbuild.files` input: https://github.com/osbuild/osbuild/blob/88da3beabc75ccff7733abe41d8e08f418d5be00/inputs/org.osbuild.files#L174-L179

Reply:
Thanks for clearing that up!
Some docs would be nice, I agree :)